Sources/FoundationEssentials/UUID.swift

```diff
@@ -76,6 +76,50 @@ public struct UUID : Hashable, Equatable, CustomStringConvertible, Sendable {
             hasher.combine(bytes: buffer)
         }
     }
+    
+    /// Generates a new random UUID.
+    ///
+    /// - Parameter generator: The random number generator to use when creating the new random value.
+    /// - Returns: A random UUID.
+    @available(FoundationPreview 6.3, *)
+    public static func random(
+        using generator: inout some RandomNumberGenerator
+    ) -> UUID {
+        let first = UInt64.random(in: .min ... .max, using: &generator)
+        let second = UInt64.random(in: .min ... .max, using: &generator)
+
+        var firstBits = first
+        var secondBits = second
+
+        // Set the version to 4 (0100 in binary)
+        firstBits &= 0xFFFFFFFFFFFF0FFF // Clear bits 48 through 51
+        firstBits |= 0x0000000000004000 // Set the version bits to '0100' at the correct position
+        
+        // Set the variant to '10' (RFC9562 variant)
+        secondBits &= 0x3FFFFFFFFFFFFFFF // Clear the 2 most significant bits
```

Comment:
```
~(0b11 << 62)
```

or maybe

```
0b11_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111
```

but of course that's so long that it's obfuscated (but not nearly as bad as `0x3FFFFFFFFFFFFFFF` IMHO)